service/utility.go

```diff
@@ -184,10 +187,33 @@ func generateToken(tokenSize int) (string, error) {
 	defer tokenLocker.Unlock()
 
 	randomBytes := make([]byte, tokenSize)
-	_, err := rand.Read(randomBytes)
+	_, err := cryptorand.Read(randomBytes)
 	if err != nil {
 		return "", err
 	}
 
 	return base64.URLEncoding.EncodeToString(randomBytes), nil
 }
+
+// generateSecretExpirationDate returns the expiration date for secret keys used for signing JWT
+// currently sets expiration date to every Monday at 3AM UTC
```

Comment:
It seems like every Monday at 3AM UTC is not good at all because if we decided to generate a secret on Sunday, then that secret only lasts for almost a day.

I suggest we should make it expire after 7 days.